catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaEventPublisher.java

```diff
@@ -0,0 +1,82 @@
+  Properties properties = new Properties();
+
+  public KafkaEventPublisher(KafkaEventConfiguration kafkaConfig) {
+    super(kafkaConfig.getBatchSize(), new ArrayList<>());
+    this.kafkaEventConfiguration = kafkaConfig;
+    if (kafkaConfig.getSecurityProtocol().equals(KafkaEventConfiguration.SecurityProtocol.SSL)) {
+      // configuration for SSL Encryption
+      if (kafkaConfig.getSSLProtocol() != null
+          && kafkaConfig.getSSLTrustStoreLocation() != null
+          && kafkaConfig.getSSLTrustStorePassword() != null
+          && kafkaConfig.getSSLKeystoreLocation() != null
+          && kafkaConfig.getSSLKeystorePassword() != null
+          && kafkaConfig.getSSLKeyPassword() != null) {
+        properties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, kafkaConfig.getSecurityProtocol().toString());
+        properties.put(SslConfigs.SSL_PROTOCOL_CONFIG, kafkaConfig.getSSLProtocol());
+        properties.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, kafkaConfig.getSSLTrustStoreLocation());
+        properties.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, kafkaConfig.getSSLTrustStorePassword());
+        // configuration for SSL Authentication
+        properties.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, kafkaConfig.getSSLKeystoreLocation());
+        properties.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, kafkaConfig.getSSLKeystorePassword());
+        properties.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, kafkaConfig.getSSLKeyPassword());
+      } else {
+        LOG.info("The SSL could not be configured as the required properties are not defined!");
+      }
+    }
+    properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaConfig.getBootstrapServers());
+    properties.put(ProducerConfig.ACKS_CONFIG, kafkaConfig.getAcks());
+    properties.put(ProducerConfig.RETRIES_CONFIG, kafkaConfig.getRetries());
+    properties.put(ProducerConfig.BATCH_SIZE_CONFIG, kafkaConfig.getBatchSize());
+    properties.put(ProducerConfig.LINGER_MS_CONFIG, kafkaConfig.getLingerMS());
+    properties.put(ProducerConfig.BUFFER_MEMORY_CONFIG, kafkaConfig.getBufferMemory());
+    properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, kafkaConfig.getKeySerializer());
+    properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, kafkaConfig.getValueSerializer());
+    producer = new KafkaProducer<>(properties);
+  }
+
+  @Override
+  public void onStart() {
+    LOG.info("Kafka Publisher Started");
+  }
+
+  @Override
+  public void onShutdown() {
+    producer.flush();
+    producer.close();
+  }
+
+  @Override
+  public void publish(EventResource.ChangeEventList events) throws JsonProcessingException {
+    if (kafkaEventConfiguration.getTopics() != null) {
+      for (String topic : kafkaEventConfiguration.getTopics()) {
+        for (ChangeEvent event : events.getData()) {
+          String eventJson = JsonUtils.pojoToJson(event);
+          producer.send(new ProducerRecord<>(topic, eventJson));
```

Comment:
this is in the path of returning a response back. Lets make sure this is async producer and we are not waiting for any ack

Reply:
The producer here is not waiting for any ack, it is directly publishing the message.